### PR TITLE
rec: Give a more clear error message if the build fails because python was not found

### PR DESCRIPTION
**Pull request by omoerbeek**
But don't report an error during configure:

When building from a src tarball you don't need python (the tarball includes the generated files), but that changes if settings/table.py was modified or when building from a github checkout.

### Checklist

I have:
- [X] read the [CONTRIBUTING.md](https://github.com/PowerDNS/pdns/blob/master/CONTRIBUTING.md) document
- [X] compiled this code
- [X] tested this code
- [ ] included documentation (including possible behaviour changes)
- [ ] documented the code
- [ ] added or modified regression test(s)
- [ ] added or modified unit test(s)

### CODE REVIEW

**Comment by paddg** [comment]
In our case python2 was installed, but python3 was needed if I remember correctly.

**Comment by omoerbeek** [comment]
> In our case python2 was installed, but python3 was needed if I remember correctly.

Ah, I'll extend the test to include that case then

**Comment by paddg** [comment]
Ah, TIL "Python 4.0 will probably never come" :)

**Comment by omoerbeek** [comment]
> Ah, TIL "Python 4.0 will probably never come" :)

I plan to be retired by then.

File: pdns/recursordist/settings/Makefile.am [hunk 1]
```diff
@@ -23,6 +23,8 @@ BUILT_SOURCES=cxxsettings-generated.cc rust/src/lib.rs
 #
 # Use patterns to avoid having two instances of generate run simultaneously, a well-known hack for GNU make
 cxxsettings-generated%cc rust/src/lib%rs: table.py generate.py rust-preamble-in.rs rust-bridge-in.rs docs-old-preamble-in.rst docs-new-preamble-in.rst
+	@if test "$(PYTHON)" = ":"; then echo "Settings table table.py has changed, need python to regenerate generated settings files but python was not found"; exit 1; fi
+	@if ! $(PYTHON) --version | grep -q "Python 3"; then echo $(PYTHON) should be at least version 3; exit 1; fi
```
Comment (zeha) [inline pdns/recursordist/settings/Makefile.am:27]: should both cases say something about needing to reconfigure/remeson?

File: pdns/recursordist/settings/Makefile.am [hunk 2]
```diff
@@ -23,6 +23,8 @@ BUILT_SOURCES=cxxsettings-generated.cc rust/src/lib.rs
 #
 # Use patterns to avoid having two instances of generate run simultaneously, a well-known hack for GNU make
 cxxsettings-generated%cc rust/src/lib%rs: table.py generate.py rust-preamble-in.rs rust-bridge-in.rs docs-old-preamble-in.rst docs-new-preamble-in.rst
+	@if test "$(PYTHON)" = ":"; then echo "Settings table table.py has changed, need python to regenerate generated settings files but python was not found"; exit 1; fi
```
Comment (rgacogne) [inline pdns/recursordist/settings/Makefile.am:26]:
```suggestion
	@if test "$(PYTHON)" = ":"; then echo "Settings table table.py has changed, python is needed to regenerate the related settings files but python was not found. Please install python and re-run configure"; exit 1; fi
```

File: pdns/recursordist/settings/Makefile.am (hunk 1 above)
Comment (rgacogne) [inline pdns/recursordist/settings/Makefile.am:27]:
```suggestion
	@if ! $(PYTHON) --version | grep -q "Python 3"; then echo $(PYTHON) should be at least version 3. Please install python 3 and re-run configure; exit 1; fi
```

**Review by rgacogne — APPROVED** [review]

**Pull request merged**